engine/execution/ingestion/loader/unfinalized_loader_test.go

```diff
@@ -0,0 +1,52 @@
+package loader_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/engine/execution/ingestion"
+	"github.com/onflow/flow-go/engine/execution/ingestion/loader"
+	stateMock "github.com/onflow/flow-go/engine/execution/state/mock"
+	"github.com/onflow/flow-go/model/flow"
+	storage "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/onflow/flow-go/utils/unittest/mocks"
+)
+
+var _ ingestion.BlockLoader = (*loader.UnfinalizedLoader)(nil)
+
+func TestLoadingUnfinalizedBlocks(t *testing.T) {
+	ps := mocks.NewProtocolState()
+
+	// Genesis <- A <- B <- C (finalized)
+	chain, result, seal := unittest.ChainFixture(4)
```

Comment:
this doesn't test pending blocks. How about making this a 5 chain, and verifying blockD is included as well?